pkg/network/server.go

```diff
@@ -404,6 +423,26 @@ func (s *Server) handleBlockCmd(p Peer, block *core.Block) error {
 	return s.bQueue.putBlock(block)
 }
 
+// handlePing processes ping request.
+func (s *Server) handlePing(p Peer, ping *payload.Ping) error {
+	return p.WriteMsg(NewMessage(s.Net, CMDPong, payload.NewPing(s.id, s.chain.BlockHeight())))
+}
+
+// handlePing processes pong request.
+func (s *Server) handlePong(p Peer, pong *payload.Ping) error {
+	if !p.PingSent() {
+		errorMsg := "pong message wasn't expected"
+		s.log.Error(errorMsg)
```

Comment:
I don't think you need to log it here, the error returned one line below will trigger a disconnect, unregistration and there is gonna be a message written with the reason of unregistration.